src/main/ruby/truffleruby/core/thread.rb

```diff
@@ -123,6 +123,12 @@ def start(...)
       thread
     end
     alias_method :fork, :start
+
+    def each_caller_location
+      raise LocalJumpError, 'no block given' unless block_given?
+
+      Primitive.thread_each_caller_location
+    end
```

Comment:
Does it make sense to have this method completely defined in Java?

Reply:
True, moving out of Ruby